src/Tempest/Support/src/ArrayHelper.php

```diff
@@ -44,6 +44,73 @@ public function __construct(
         }
     }
 
+    /**
+     * Search for a given value in the array and return the corresponding key if successful.
+     *
+     * @param Closure|mixed $value The value to search for, a Closure will check the first item that returns true.
+     * @param bool $strict Whether to use strict comparison.
+     *
+     * @return array-key|false The key for `$value` if found, otherwise `false`.
+     */
+    public function search(mixed $value, bool $strict = false): int|string|false
```

Comment:
This method is a wrapper around https://www.php.net/manual/en/function.array-search.php so I don't think this should returns a value.
But I probably have misunderstood the question ?

Reply:
Yea, my bad, it's not the same thing—I feel like the method name should be something like `findKey`, `searchKey`, or something like that instead though

Reply:
Why not, or maybe an alias as I've done multiple times in this helper.
Cause I've taken 2 main ideas :
- Keep a logic with other similar helpers (e.g. Laravel collections which have a `search` method aswell with the same behavior https://laravel.com/docs/11.x/collections#method-search )
- Stay as consistent as possible with the builtin array functions as this is the goal of this helper, so `array_search` returns a key

Then I'd go for keeping the signature like that and add alias to it like `findKey`

WDYT ?

Reply:
Well personally I'd rather have a clear method name, which neither the built-in PHP function nor the Laravel equivalent have

I don't have a strong opinion on it though, it's just about clarity for someone who'd be looking for something through the method names—the more explicit, the better, and I don't think that just `search` conveys what it does

Reply:
`search` is a very confusing name. I feel like we can diverge from PHP whenever PHP does weird things. I think `findKey` makes the most sense, since we also have `find` for values.

I'd also rename the parameter for extra clarity: `findKey(mixed $value)`

Reply:
Also, instead of returning `false` when there's no hit, let's return `null`.

Reply:
Ok let's do that 👍 

false seems weird to me but it was to be consistent with the php/laravel implementation.

Reply:
@brendt We actually don't have `find` method
If this is wanted, please tell me, I'll add it to the TODO list on the issue

Reply:
Ah well, we'll add it when PHP 8.4 comes around, since `array_find` is added there as well :) You can add it to the todo list :)